perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -594,53 +590,56 @@ ObjectClassification::_label_type changeLabelForPrediction(
-    const bool within_road_lanelet = withinRoadLanelet(object, lanelet_map_ptr_, true);
-    const float high_speed_threshold = 25.0 / 18.0 * 5.0;  // High speed bicycle 25 km/h
-    // calc abs speed from x and y velocity
-    const double abs_speed = std::hypot(
-      object.kinematics.twist_with_covariance.twist.linear.x,
-      object.kinematics.twist_with_covariance.twist.linear.y);
-    const bool high_speed_object = abs_speed > high_speed_threshold;
-
-    // if the object is within lanelet, do the same estimation with vehicle
-    if (within_road_lanelet) {
-      return ObjectClassification::MOTORCYCLE;
-    } else if (high_speed_object) {
+  switch (label) {
+    case ObjectClassification::CAR:
+    case ObjectClassification::BUS:
+    case ObjectClassification::TRUCK:
+    case ObjectClassification::TRAILER:
+    case ObjectClassification::UNKNOWN:
+      return label;
+      break;
+
+    case ObjectClassification::MOTORCYCLE:
+    case ObjectClassification::BICYCLE: {  // if object is within road lanelet and satisfies yaw
+                                           // constraints
+      const bool within_road_lanelet = withinRoadLanelet(object, lanelet_map_ptr_, true);
+      const float high_speed_threshold = 25.0 / 18.0 * 5.0;  // High speed bicycle 25 km/h
+      // calc abs speed from x and y velocity
+      const double abs_speed = std::hypot(
+        object.kinematics.twist_with_covariance.twist.linear.x,
+        object.kinematics.twist_with_covariance.twist.linear.y);
+      const bool high_speed_object = abs_speed > high_speed_threshold;
+
+      // if the object is within lanelet, do the same estimation with vehicle
+      if (within_road_lanelet) return ObjectClassification::MOTORCYCLE;
       // high speed object outside road lanelet will move like unknown object
       // return ObjectClassification::UNKNOWN; // temporary disabled
-      return label;
-    } else {
+      if (high_speed_object) return label;  // Do nothing for now
       return ObjectClassification::BICYCLE;
+      break;
     }
-  } else if (label == ObjectClassification::PEDESTRIAN) {
-    const bool within_road_lanelet = withinRoadLanelet(object, lanelet_map_ptr_, true);
-    const float max_velocity_for_human_mps =
-      25.0 / 18.0 * 5.0;  // Max human being motion speed is 25km/h
-    const double abs_speed = std::hypot(
-      object.kinematics.twist_with_covariance.twist.linear.x,
-      object.kinematics.twist_with_covariance.twist.linear.y);
-    const bool high_speed_object = abs_speed > max_velocity_for_human_mps;
-    // fast, human-like object: like segway
-    if (within_road_lanelet && high_speed_object) {
-      return label;  // currently do nothing
+
+    case ObjectClassification::PEDESTRIAN: {
+      const bool within_road_lanelet = withinRoadLanelet(object, lanelet_map_ptr_, true);
+      const float max_velocity_for_human_mps =
+        25.0 / 18.0 * 5.0;  // Max human being motion speed is 25km/h
```

Comment:
Same comment here
```
tier4_autoware_utils::kmph2mps(25.0)
```

Reply:
Thank you. Will look into it once I have time.

Reply:
@YoshiRi  Done